mllib/src/main/scala/org/apache/spark/ml/feature/MinHashLSH.scala

```diff
@@ -74,9 +72,12 @@ class MinHashModel private[ml] (
   }
 
   @Since("2.1.0")
-  override protected[ml] def hashDistance(x: Vector, y: Vector): Double = {
+  override protected[ml] def hashDistance(x: Seq[Vector], y: Seq[Vector]): Double = {
     // Since it's generated by hashing, it will be a pair of dense vectors.
-    x.toDense.values.zip(y.toDense.values).map(pair => math.abs(pair._1 - pair._2)).min
+    // TODO: This hashDistance function is controversial. Requires more discussion.
+    x.zip(y).map(vectorPair =>
```

Comment:
Since it's still under discussion, I am not sure which `hashDistance` to leave in the code. Do you just want me to change to the `hashDistance` @jkbradley suggested?